clicache/src/CMakeLists.txt

```diff
@@ -48,21 +49,23 @@ target_link_libraries(Apache.Geode
 
 string(REPLACE "/EHsc" "/EHa" CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}")
 string(REPLACE "/RTC1" "" CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG}")
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /clr /wd4947 /wd4251 /wd4635 /doc /we4488")
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /clr /bigobj /wd4947 /wd4251 /wd4635 /doc /we4488")
 set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} ${SHARED_LINKER_FLAGS_STRONG_KEY}")
 
-include_directories(${CMAKE_SOURCE_DIR}/clicache/include)
-include_directories(${CMAKE_SOURCE_DIR}/clicache/src)
 include_directories(${CMAKE_SOURCE_DIR}/cppcache/src)
 
-set_target_properties(Apache.Geode PROPERTIES
+set_target_properties(${PROJECT_NAME} PROPERTIES
   OUTPUT_NAME ${PRODUCT_DLL_NAME}
   VS_DOTNET_TARGET_FRAMEWORK_VERSION "v4.5.2"
-  VS_DOTNET_REFERENCES "System;System.Xml")
+  VS_DOTNET_REFERENCES "System;System.Xml"
+)
+
+include(PrecompiledHeader)
+add_precompiled_header(${PROJECT_NAME} geode_includes.hpp FORCEINCLUDE)
```

Comment:
Added precompiled headers to work around and speed up issue including marshal_as headers.